### ISSUE TITLE
Having problems loading parameters

### ISSUE THREAD

**Issue by lzuwei** [dnouri/nolearn#276]
Hi all, I am having some problems loading models properly.

I loaded a pretrained model from matlab and did some fine tuning and saved the fine tuned model for future use. Please look at the following:

```
def load_dbn(path='../../DBNExample/pretrained_nn.mat'):
    """
    load a pretrained dbn from path
    :param path: path to the .mat dbn
    :return: pretrained deep belief network
    """
    # create the network using weights from pretrain_nn.mat
    nn = sio.loadmat(path)
    w1 = nn['w1']
    w2 = nn['w2']
    w3 = nn['w3']
    w4 = nn['w4']
    w5 = nn['w5']
    w6 = nn['w6']
    w7 = nn['w7']
    w8 = nn['w8']
    b1 = nn['b1'][0]
    b2 = nn['b2'][0]
    b3 = nn['b3'][0]
    b4 = nn['b4'][0]
    b5 = nn['b5'][0]
    b6 = nn['b6'][0]
    b7 = nn['b7'][0]
    b8 = nn['b8'][0]

    layers = [
        (InputLayer, {'name': 'input', 'shape': (None, 1200)}),
        (DenseLayer, {'name': 'l1', 'num_units': 2000, 'nonlinearity': sigmoid, 'W': w1, 'b': b1}),
        (DenseLayer, {'name': 'l2', 'num_units': 1000, 'nonlinearity': sigmoid, 'W': w2, 'b': b2}),
        (DenseLayer, {'name': 'l3', 'num_units': 500, 'nonlinearity': sigmoid, 'W': w3, 'b': b3}),
        (DenseLayer, {'name': 'l4', 'num_units': 50, 'nonlinearity': linear, 'W': w4, 'b': b4}),
        (DenseLayer, {'name': 'l5', 'num_units': 500, 'nonlinearity': sigmoid, 'W': w5, 'b': b5}),
        (DenseLayer, {'name': 'l6', 'num_units': 1000, 'nonlinearity': sigmoid, 'W': w6, 'b': b6}),
        (DenseLayer, {'name': 'l7', 'num_units': 2000, 'nonlinearity': sigmoid, 'W': w7, 'b': b7}),
        (DenseLayer, {'name': 'output', 'num_units': 1200, 'nonlinearity': linear, 'W': w8, 'b': b8}),
    ]

    dbn = NeuralNet(
        layers=layers,
        max_epochs=30,
        objective_loss_function=squared_error,
        update=nesterov_momentum,
        regression=True,
        verbose=1,
        update_learning_rate=0.001,
        update_momentum=0.05,
        objective_l2=0.005,
    )
    return dbn
```

Next I trained the model further for say 30 epochs using ````dbn.fit(X, y)``` and saved it using ```dbn.save_params_to('path/to/model')```. However when I tried to load the model using ```dbn.load_params_from('path/to/model/') with the following code:

```
def load_finetuned_dbn(path):
    """
    Load a fine tuned Deep Belief Net from file
    :param path: path to deep belief net parameters
    :return: deep belief net
    """
    dbn = NeuralNet(
        layers=[
            ('input', las.layers.InputLayer),
            ('l1', las.layers.DenseLayer),
            ('l2', las.layers.DenseLayer),
            ('l3', las.layers.DenseLayer),
            ('l4', las.layers.DenseLayer),
            ('l5', las.layers.DenseLayer),
            ('l6', las.layers.DenseLayer),
            ('l7', las.layers.DenseLayer),
            ('output', las.layers.DenseLayer)
        ],
        input_shape=(None, 1200),
        l1_num_units=2000, l1_nonlinearity=sigmoid,
        l2_num_units=1000, l2_nonlinearity=sigmoid,
        l3_num_units=500, l3_nonlinearity=sigmoid,
        l4_num_units=50, l4_nonlinearity=linear,
        l5_num_units=500, l5_nonlinearity=sigmoid,
        l6_num_units=1000, l6_nonlinearity=sigmoid,
        l7_num_units=2000, l7_nonlinearity=sigmoid,
        output_num_units=1200, output_nonlinearity=linear,
        update=nesterov_momentum,
        update_learning_rate=0.001,
        update_momentum=0.5,
        objective_l2=0.005,
        verbose=1
    )
    with open(path, 'rb') as f:
        pretrained_nn = pickle.load(f)
    if pretrained_nn is not None:
        dbn.load_params_from(path)
    return dbn
```

The model loads successfully with the following output:

Loaded parameters to layer 'l1' (shape 1200x2000).
Loaded parameters to layer 'l1' (shape 2000).
Loaded parameters to layer 'l2' (shape 2000x1000).
Loaded parameters to layer 'l2' (shape 1000).
Loaded parameters to layer 'l3' (shape 1000x500).
Loaded parameters to layer 'l3' (shape 500).
Loaded parameters to layer 'l4' (shape 500x50).
Loaded parameters to layer 'l4' (shape 50).
Loaded parameters to layer 'l5' (shape 50x500).
Loaded parameters to layer 'l5' (shape 500).
Loaded parameters to layer 'l6' (shape 500x1000).
Loaded parameters to layer 'l6' (shape 1000).
Loaded parameters to layer 'l7' (shape 1000x2000).
Loaded parameters to layer 'l7' (shape 2000).
Loaded parameters to layer 'output' (shape 2000x1200).
Loaded parameters to layer 'output' (shape 1200).

which seems correct. But when testing it with ```predictX = dbn.predict(test_X)```  where ```test_X``` is of shape (6230, 1200), the output of the predictions are of the incorrect dimensions. I expect the output to be of shape (6230, 1200) but I obtained a shape of (6230,) of int64 instead.

Any ideas?

**Comment by dnouri** [comment]
By the way there's also net.predict_proba that'll output the probabilities
for each class.

**Comment by lzuwei** [comment]
Thank you for the pointers, I didn't realize the regression option wasn't set which turned it into a classifier.
